File: Java/src/main/java/QuestionBank/Solution601_700/Solution605.java
package QuestionBank.Solution601_700;

/**
 * @author windows
 */
public class Solution605 {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        for (int i = 0, len = flowerbed.length; i < len && n > 0;) {
            if (flowerbed[i] == 1) {
                i += 2;
            } else if (i == flowerbed.length - 1 || flowerbed[i + 1] == 0) {
                n--;
                i += 2;
            } else {
                i += 3;
            }
        }
        return n <= 0;
    }

    public static void main(String[] args) {
        Solution605 solution605 = new Solution605();
        System.out.println(solution605.canPlaceFlowers(new int[]{1,0,0,0,1,0,0}, 2));
    }
}

File: C++/src/Solution601_700/Solution605.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution605.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

bool Solution605::canPlaceFlowers(vector<int> &flowerbed, int n) {
    for (int i = 0; i < flowerbed.size(); ++i) {
        if (flowerbed[i] == 0 && (i - 1 < 0 ||(i - 1 > 0 && flowerbed[i - 1] == 0)) && ((i + 1 < flowerbed.size() && flowerbed[i + 1] == 0) || i + 1 >= flowerbed.size())) {
            flowerbed[i] = 1;
            n--;
        }
    }
    return n <= 0;
}
